### DISCUSSION TITLE
Multiple User Times & Export on Tablets (eg. iPad)

### DISCUSSION THREAD

**Discussion by empx92** [kimai/kimai#5060] [General]
Description of Usecase "Multiple User Times":
If a team on the construction site consists of, for example, a foreman, an apprentice, and an intern, ideally the foreman should be able to log time for the intern or apprentice. This already works under "All Times," but the user "foreman" needs the function to edit times of others, which is a disadvantage because they should only be able to edit the times of others that they have added themselves. In case of an incorrect entry, the foreman cannot change the time of their employees anymore and has to contact the office (administrator). Additionally, it would be beneficial if the work description could be entered specifically per user (if needed) so that the export does not show the same description three times. Nice thing too would be if the foreman could write the times inside his calender (not all times) and if there could be a feature where you can see that this time block exists for other users or even can edit it when clicking on one of them. We are using this every day but our administration always has to go through all users if there is one small mistake or if they choose a false project by accident. 
I know that there may be some functionalities by day, but in a digital transformation process it's all about a easy workflow for our employees ;)

Regarding the iPad: 
When adding Kimai to the home screen via Safari on the iPad, the PDF can be displayed during export, but it cannot be saved to the local files, and it’s not possible to return to the dashboard. 
The background is that the user, for example, should be able to export a project at the client’s location and have it signed by the client right away. Alternatively, a signature field that remains valid until a time entry is changed would be more convenient, but as discussed, this is almost impossible to implement.
As the PDF is just generated when clicking on the button, i don't really know how it could be solved. Just one thing i saw in another PWA is, that they are using Modals for .pdf and some kind of "open in new window" which seems to work.

Did anyone run in to same Issues or found a workaround for that?

**Comment by kevinpapst** [comment]
Can you sort your thoughts and add multiple, clear structured, posts?
This is a nightmare to read and understand.

Regarding the iPad: this is an Apple issue. I don't think I can fix that.
